crates/common/src/raindex_client/orderbook_yaml.rs

```diff
@@ -0,0 +1,339 @@
+    /// if (result.error) {
+    ///   console.error("Error getting networks:", result.error.readableMsg);
+    ///   return;
+    /// }
+    /// const networks = result.value;
+    /// for (const [key, config] of networks) {
+    ///   console.log(`Network key: ${key}, Chain ID: ${config.chainId}`);
+    /// }
+    /// ```
+    #[wasm_export(
+        js_name = "getAllNetworks",
+        return_description = "Returns a map of all available networks with their configurations. Keys are network names, values are NetworkCfg objects.",
+        unchecked_return_type = "Map<string, NetworkCfg>"
+    )]
+    pub fn get_all_networks(&self) -> Result<HashMap<String, NetworkCfg>, RaindexError> {
+        Ok(self.orderbook_yaml.get_networks()?)
+    }
+
+    /// Retrieves network configuration for a specific chain ID
+    ///
+    /// Finds and returns the network configuration that matches the provided chain ID.
+    /// This is useful when you need to access network-specific settings like RPC URLs.
+    ///
+    /// ## Examples
+    ///
+    /// ```javascript
+    /// const result = client.getNetworkByChainId(1); // Ethereum mainnet
+    /// if (result.error) {
+    ///   console.error("Network not found:", result.error.readableMsg);
+    ///   return;
+    /// }
+    /// const networkConfig = result.value;
+    /// console.log(`Found network: ${networkConfig}`);
+    /// ```
+    #[wasm_export(
+        js_name = "getNetworkByChainId",
+        return_description = "Returns the configuration for a specific network identified by its chain ID",
+        unchecked_return_type = "NetworkCfg"
+    )]
+    pub fn get_network_by_chain_id(
+        &self,
+        #[wasm_export(
+            js_name = "chainId",
+            param_description = "The blockchain network ID to retrieve the configuration for"
+        )]
+        chain_id: u32,
+    ) -> Result<NetworkCfg, RaindexError> {
+        Ok(self.orderbook_yaml.get_network_by_chain_id(chain_id)?)
+    }
+
+    /// Retrieves orderbook configuration by contract address
+    ///
+    /// Finds and returns the orderbook configuration that matches the provided contract address.
+    /// This allows you to access orderbook-specific settings including subgraph endpoints,
+    /// network information, and other details.
+    ///
+    /// ## Examples
+    ///
+    /// ```javascript
+    /// const result = client.getOrderbookByAddress("0x1234567890123456789012345678901234567890");
+    /// if (result.error) {
+    ///   console.error("Orderbook not found:", result.error.readableMsg);
+    ///   return;
+    /// }
+    /// const orderbookConfig = result.value;
+    /// console.log(`Found orderbook ${orderbookConfig}`);
+    /// ```
+    #[wasm_export(
+        js_name = "getOrderbookByAddress",
+        return_description = "Returns the configuration for a specific orderbook identified by its address",
+        unchecked_return_type = "OrderbookCfg"
+    )]
+    pub fn get_orderbook_by_address_wasm_binding(
```

Comment:
also nothing that says you can't have two orderbooks in the orderbooks map with the same address